dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/hibernate/HibernateTrackedEntityInstanceStore.java

```diff
@@ -484,11 +486,55 @@ private String getQuerySelect( TrackedEntityInstanceQueryParams params, boolean
         {
             select.append( ", TEI.trackedentityinstanceid AS teiid" );
         }
+
+        if ( params.getOrders() != null )
+        {
+            select.append( getOrderByQuery( params, select ) );
+        }
+
         select.append( SPACE );
 
         return select.toString();
     }
 
+    private String getOrderByQuery( TrackedEntityInstanceQueryParams params, StringBuilder select )
+    {
+        StringBuilder orderQuery = new StringBuilder();
+
+        for ( OrderParam orderParam : params.getOrders() )
+        {
+            if ( isStaticColumn( orderParam.getField() ) )
+            {
+                if ( ENROLLED_AT.getPropName().equalsIgnoreCase( orderParam.getField() ) )
+                {
+                    //In the main query, we need to use the alias "tei" to fetch the value of the field enrolledAt
+                    orderQuery
+                        .append( addStaticColumn( orderParam.getField(), select.toString() ).replace( "pi", "tei" ) );
```

Comment:
Do I understand correctly that if someone renames the alias in one place (i.e. main query) but not the other we will have a bug?

If so we should tie the alias from the main query together with the alias used here. For example via constant.